include/h2o/memory.h

```diff
@@ -301,6 +305,13 @@ void h2o_append_to_null_terminated_list(void ***list, void *element);
 
 /* inline defs */
 
+inline void *h2o_memcpy(void *dst, const void *src, size_t n)
+{
+    if (src)
```

Comment:
I don't think we can do that, because the compiler would see `src` being passed to `memcpy`, then assume `src` cannot be `NULL`, then remove as much `NULL` checks for `src` that it can. So my understanding is that we must specifically guard against `src` being `NULL`.

I believe that ubsan is warning us against is a scenario along the lines of what's decribed in 2.5 in this paper: https://people.csail.mit.edu/nickolai/papers/wang-undef-2012-08-21.pdf . What do you think?